app/pages/_app.js

```diff
@@ -2,13 +2,16 @@ import '/styles/globals.css';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import Layout from '../components/layout';
 import SSRProvider from 'react-bootstrap/SSRProvider';
+import AuthProvider from '../components/auth/firebase/AuthProvider';
```

Comment:
We *really* don't want global hooks at this level if it can be avoided.

It is architecture changing and will impact  SSR vs static vs client strategies. 

Investigate use of the  NextJS  recommended   `next-firebase-auth` module.

Reply:
Added firebase-next-auth.
The page component needs to be exported with `withAuthUser` or `withAuthUserSSR`  and then useAuthUser( ) can be used to retrieve user information.